configuration/haproxy.go

```diff
@@ -8,4 +10,29 @@ type HAProxy struct {
 	ReloadCleanupCommand    string
 	ShutdownCommand         string
 	GraceSeconds            int
+	HostnameLabel           *string
+}
+
+// BalancerType indicates whether there bamboo is dealing with external or internal traffic
+type BalancerType string
```

Comment:
Now if possible because I think taking arbitrary balancer names would simplify this code a bit?